### ISSUE TITLE
Project Name not set on Google Cloud SQL Database

### ISSUE THREAD

**Issue by arhea** [terraform-providers/terraform-provider-google#233]
### Terraform Version
`0.10.0-rc1`

### Affected Resource(s)
Please list the resources as a list, for example:
- `google_sql_database`

### Terraform Configuration Files
```hcl
resource "google_sql_database" "mydb" {
  depends_on = [
    "null_resource.is_ready",
    "google_sql_database_instance.mydb",
    "google_sql_user.mydb",
  ]

  name      = "mydb"
  instance  = "${google_sql_database_instance.mydb.name}"
  charset   = "UTF8"
  collation = "en_US.UTF8"
}
```

### Debug Output
```
module.bitbucket.google_sql_database.mydb: Creating...
  charset:   "" => "UTF8"
  collation: "" => "en_US.UTF8"
  instance:  "" => "mydb-db-2f20fd5d11b44799"
  name:      "" => "mydb"
  self_link: "" => "<computed>"
```

### Steps to Reproduce
1. `terraform apply`

### Important Factoids
The error returned from the Google API is `sql#operation INTERNAL_ERROR`. After debugging by creating a database through the UI, which worked we determined that the missing field on the request was the `project` field. Without this field explicitly set the API fails. The work around is to set the project manually as such.

It is also worth mentioning that the `collation` defaults to a MySQL value which may lead to weird errors with PostgreSQL databases.

```hcl
resource "google_sql_database" "mydb" {
  depends_on = [
    "null_resource.is_ready",
    "google_sql_database_instance.mydb",
    "google_sql_user.mydb",
  ]

  name       = "mydb"
  instance  = "${google_sql_database_instance.mydb.name}"
  project    = "my-project"
  charset    = "UTF8"
  collation   = "en_US.UTF8"
}
```

**Comment by ubschmidt2** [comment]
This might be the same as https://github.com/terraform-providers/terraform-provider-google/issues/167.

Regarding the default for charset and collation: This is a very good point, PostgreSQL wasn't on my radar. I'm proposing https://github.com/terraform-providers/terraform-provider-google/pull/229.

**Comment by arhea** [comment]
@ubschmidt2 - I believe #167 might be related as well. Google API likes to have the project explicitly set it looks like.

#229 is definitely a step in the right direction. Support / Documentation for PostgreSQL is needed.

**Comment by ubschmidt2** [comment]
@arhea - PostgreSQL is already supported and documented (https://github.com/terraform-providers/terraform-provider-google/commit/25c6a00ca4388af1ae05ce5e9ea23adfe0b0336c, https://github.com/hashicorp/terraform/issues/12617).

**Comment by ubschmidt2** [comment]
@arhea - I've added minimal documentation to #229.

I've created a separate issue for improvements to the charset/collation support for PostgreSQL, https://github.com/terraform-providers/terraform-provider-google/issues/260 as this issue is primarily about the missing project id.

**Comment by danawillow** [comment]
Just like in #167, I can't seem to reproduce this :-/

Can you share your full config instead of just the affected resource? (I really just need the `provider` block but anything else you can share would be great) Also full debug logs could help- set the environment variable `TF_LOG=DEBUG`

**Comment by danawillow** [comment]
I think this was fixed in #469. Let me know if it's still happening and I'll reopen!
